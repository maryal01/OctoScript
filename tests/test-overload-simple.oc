fn overload printout(int i) -> none {
    print("%d", i);
}

fn overload printout(string s) -> none {
    print("%s", s);
}

printout(42);
printout("hello world");

fn overload foo(int i) -> int {
    return i;
}

fn overload foo(string s) -> string {
    return s;
}

print("%d %s", foo(42), foo("overloading is fun"));