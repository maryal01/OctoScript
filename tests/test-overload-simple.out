42hello world42 overloading is fun